### PR TITLE
[Azure.Communication.JobRouter] Initial commit (#29251)

### PR DESCRIPTION
**Pull request by sarkar-rajarshi**
* JobRouter - initial commit

* test autorest codegen with ci

* bugfix (#26478)

* JobRouter readme file (#26516)

readme

* Fix broken test + add samples for QueueSelectors in classification policy (#27001)

* Fix broken test + add sample for QueueSelectors in classification policy

* incorporate PR comments

* pr comments

* pr comments

* Add queueing scenarios tests (#27053)

* Fix broken test + add sample for QueueSelectors in classification policy

* Add different queueing scenarios

* rebase with jobRouter/initial-commit

* additional overloads (#27055)

* JobRouter - initial commit

* initial commit

* initial commit of preview 2

Co-authored-by: Daniel Gerlag <daniel@gerlag.ca>

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs [hunk 1]
```diff
@@ -0,0 +1,1345 @@
...
+        #endregion
+
+        #region private constructors
+
+        private RouterClient(ConnectionString connectionString, RouterClientOptions options)
+            : this(connectionString.GetRequired("endpoint"), options.BuildHttpPipeline(connectionString), options)
+        {
+        }
+
+        private RouterClient(string endpoint, TokenCredential tokenCredential, RouterClientOptions options)
```
Comment (ShivangiReja) [inline sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs:79]: nit: rename tokenCredential -> credential

File: sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs [hunk 2]
```diff
@@ -0,0 +1,1345 @@
...
+            : this(endpoint, options.BuildHttpPipeline(tokenCredential), options)
+        {
+        }
+
+        private RouterClient(string endpoint, AzureKeyCredential keyCredential, RouterClientOptions options)
```
Comment (ShivangiReja) [inline sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs:84]: nit: rename keyCredential -> credential

File: sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs [hunk 3]
```diff
@@ -0,0 +1,1345 @@
...
+        }
+
+        #endregion
+
+        /// <summary>Initializes a new instance of <see cref="RouterClient"/> for mocking.</summary>
+        protected RouterClient()
+        {
+            _clientDiagnostics = null;
+            RestClient = null;
+        }
```
Comment (ShivangiReja) [inline sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs:102]: Can we move this at the top with constructors section?

File: sdk/communication/Azure.Communication.JobRouter/src/autorest.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+3. Repeat 2 and 3 until the decided interface is reflected in the apiview.dev 
+
+> see [https://aka.ms/autorest](https://aka.ms/autorest)
+
+## Configuration
+
+```yaml
+tag: package-jobrouter-2022-07-18-preview
+model-namespace: false
```
Comment (ShivangiReja) [inline sdk/communication/Azure.Communication.JobRouter/src/autorest.md:16]: Since we have a lot of models in this library, can we generate all the models in `Azure.Communication.JobRouter.Models` namespace?

If you remove `model-namespace: false` and regenerate code again, then all the models will be generated in `Azure.Communication.JobRouter.Models` namespace.

**Comment by ShivangiReja** [comment]
@sarkar-rajarshi can we add public APIs by running [eng\scripts\Export-API.ps1](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions) command? this command will generate a folder like [this](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/communication/Azure.Communication.Chat/api).

File: sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+## 1.0.0-beta.1 (2021-10-20)
+This is the beta release of Azure Communication Router .NET SDK. For more information, please see the [README][read_me].
+
+This is a private Preview version, so breaking changes are possible in subsequent releases as we improve the product. To provide feedback, please submit an issue in our [Azure SDK for .NET GitHub repo][issues].
+
+
+<!-- LINKS -->
+<!-- [read_me]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.JobRouter/README.md -->
+<!-- [issues]: https://github.com/Azure/azure-sdk-for-net-pr/issues -->
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md:9]:
```suggestion
<!-- [issues]: https://github.com/Azure/azure-sdk-for-net/issues -->
```

File: sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+## 1.0.0-beta.1 (2021-10-20)
+This is the beta release of Azure Communication Router .NET SDK. For more information, please see the [README][read_me].
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md:2]: Neither of these links work due to them being enclosed in comments.

File: sdk/communication/Azure.Communication.JobRouter/README.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
+# Azure Communication JobRouter client library for .NET
+
+This package contains a C# SDK for Azure Communication Services for JobRouter.
+
+[Source code][source] | [Package (NuGet)]<!-- [https://www.nuget.org/packages/Azure.Communication.JobRouter]--> | [Product documentation][product_docs]
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/README.md:5]: This renders incorrectly due to the comment.   I'd recommend routing this to the NuGet home page until you're published.    _(I'd also suggest a tracking issue to remember to set the actual link before you publish)_

File: sdk/communication/Azure.Communication.JobRouter/README.md [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+
+### Using statements
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_UsingStatements
+using Azure.Communication.JobRouter;
+```
+
+### Create a JobRouter Client
+
+This will allow you to interact with the JobRouter Service
+```C#
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/README.md:32]: This should be a snippet.

File: sdk/communication/Azure.Communication.JobRouter/README.md [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+
+### Offer
+We should get a [RouterWorkerOfferIssued][offer_issued_event_schema] from our [EventGrid subscription][subscribe_events].
+
+There are several different Azure services that act as a [event handler][event_grid_event_handlers].
+For this scenario, we are going to assume Webhooks for event delivery. [Learn more about Webhook event delivery][webhook_event_grid_event_delivery]
+
+Once events are delivered to the event handler, we can deserialize the JSON payload into a list of events.
+
+```C#
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/README.md:155]: These samples should be injected via snippet.

File: sdk/communication/Azure.Communication.JobRouter/README.md [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+                Console.WriteLine(egEvent.Data.ToString());
+                break;
+        }
+    }
+}
+```
+
+However, we could also wait a few seconds and then query the worker directly against the JobRouter API to see if an offer was issued to it.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_QueryWorker_Async
+var result = await routerClient.GetWorkerAsync(worker.Value.Id);
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/README.md:194]: We generally encourage avoiding `var` for sample snippets unless it's a direct `new` assignment, since there's no way to inspect the code and understand the type.

File: sdk/communication/Azure.Communication.JobRouter/README.md [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+}
+```
+
+### Accept an offer
+Once a worker receives an offer, it can take two possible actions: accept or decline. We are going to accept the offer.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_AcceptOffer_Async
+// fetching the offer id
+var jobOffer = result.Value.Offers.FirstOrDefault(x => x.JobId == job.Value.Id);
+
+var offerId = jobOffer!.Id; // `OfferId` can be retrieved directly from consuming event from Event grid
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/README.md:207]: We generally encourage avoiding nullable reference use in sample snippets.

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 10 above)
Comment (jsquire) [reply sdk/communication/Azure.Communication.JobRouter/README.md:194]: Leaving the one comment here, but this pattern exists in many of the other samples as well.

File: sdk/communication/Azure.Communication.JobRouter/samples/ExceptionPolicyCrud.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+var routerAdministrationClient = new RouterAdministrationClient(Environment.GetEnvironmentVariable("AZURE_COMMUNICATION_SERVICE_CONNECTION_STRING"));
+```
+
+## Create an exception policy
+
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_Crud_CreateExceptionPolicy
+var exceptionPolicyId = "my-exception-policy";
+
+// we are going to create 2 rules:
+// 1. `EscalateJobOnQueueOverFlowTrigger`: triggers when queue has more than 10 jobs already en-queued,
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/samples/ExceptionPolicyCrud.md:24]: Including markdown formatting in a code snippet is odd.   Not going to hurt anything, but they render as visual noise in a code block.

File: sdk/communication/Azure.Communication.JobRouter/samples/JobQueueCrud.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_UsingStatements
+using Azure.Communication.JobRouter;
+```
+
+## Create a client
+
+Create a `RouterClient`.
+
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_CreateClient
+var routerClient = new RouterClient(Environment.GetEnvironmentVariable("AZURE_COMMUNICATION_SERVICE_CONNECTION_STRING"));
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/samples/JobQueueCrud.md:14]: We generally use placeholder slugs for connection strings in samples rather than complicate things by introducing environment variables.   Example:

```csharp
var routerClient = new RouterClient("<< CONNECTION STRING >>");
```

File: sdk/communication/Azure.Communication.JobRouter/samples/README.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
+---
+page_type: sample
+languages:
+- csharp
+products:
+- azure
+- azure-communication-services
+# - azure-communication-services-job-router
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/samples/README.md:8]: We should consider uncommenting this or removing it.

File: sdk/communication/Azure.Communication.JobRouter/samples/README.md [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+    - [Dynamically assigning priority to job][prioritizationWithClassificationPolicyAsync]
+    - [Dynamically attach WorkerSelectors to job][attachedWorkerSelectorWithClassificationPolicyAsync]
+  - Using Distribution Policy
+    - [Basic Scenario][distributingOffersSimpleAsync]
+    - [Multiple offers for a job][distributingOffersAdvancedAsync]
+  - Using Exception Policy
+    - [Trigger exception with WaitTimeExceptionTrigger][waitTimeExceptionTriggerAsync]
+    - [Trigger exception with QueueLengthExceptionTrigger][queueLengthExceptionTriggerAsync]
+
+
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/samples/README.md:38]:
```suggestion

```

File: sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/ClassificationPolicyLiveTests.cs [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+using System.Threading.Tasks;
+using Azure.Communication.JobRouter.Tests.Infrastructure;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Communication.JobRouter.Tests.RouterClients
+{
+    public class ClassificationPolicyLiveTests : RouterLiveTestBase
+    {
+        /// <inheritdoc />
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/ClassificationPolicyLiveTests.cs:17]: I don't believe this is needed...

File: sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/DistributionPolicyLiveTests.cs [hunk 17]
```diff
@@ -0,0 +1,257 @@
...
+
+            paramSelectors = ((BestWorkerMode)bestWorkerModeDistributionPolicy.Mode).ScoringRuleOptions
+                .ScoringParameters;
+            Assert.AreEqual(1, paramSelectors.Count);
+            Assert.AreEqual(ScoringRuleParameterSelector.WorkerSelectors, paramSelectors.First());
+
+            scoringRule = ((BestWorkerMode)bestWorkerModeDistributionPolicy.Mode).ScoringRule;
+            Assert.NotNull(scoringRule);
+            azureFuncScoringRule = (FunctionRule)scoringRule;
+            // Assert.AreEqual("https://my.function.app/api/myfunction?code=Kg==", azureFuncScoringRule.FunctionAppUrl);
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/DistributionPolicyLiveTests.cs:154]: nit:  If this is unused, we should consider removing it.

File: sdk/communication/Azure.Communication.JobRouter/tests/Samples/ClassificationPolicyCrudOps.cs [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+    {
+        [Test]
+        public void ClassificationPolicyCrud()
+        {
+#if !SNIPPET
+            // create a client
+            var routerAdministrationClient = new RouterAdministrationClient(Environment.GetEnvironmentVariable("AZURE_COMMUNICATION_SERVICE_CONNECTION_STRING"));
+#endif
+
+            #region Snippet:Azure_Communication_JobRouter_Tests_Samples_Crud_CreateClassificationPolicy
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/tests/Samples/ClassificationPolicyCrudOps.cs:25]: This snippet won't compile due to the above `#if` block.   Since that conditional is outside the scope of the snippet region, I don't think we need the conditional there.

File: sdk/communication/ci.yml [hunk 19]
```diff
@@ -26,6 +26,8 @@ extends:
     ServiceDirectory: communication
     ArtifactName: packages
```
Comment (jsquire) [inline sdk/communication/ci.yml:27]: Please include the directive ` BuildSnippets: true` to ensure that snippets are compiled.     [Example](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/ci.yml#L41)

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 7 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md:5]: https://github.com/Azure/azure-sdk-for-net/issues/30319

File: sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md (hunk 6 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md:5]: The read_me link will be broken until the PR is merged to the main branch. Will be pushing in a second PR uncommenting the link section

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 9 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md:157]: This particular sample is a duplicate of https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/eventgrid/Azure.Messaging.EventGrid#receiving-and-deserializing-events

In this case, we are specifying that this is _one of the way_ to consume the event from EventHub. If this snippet is to be injected, we'll need to take a dependency on event grid sdk.

File: sdk/communication/Azure.Communication.JobRouter/README.md [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+    assignmentId: acceptJobOfferResult.Value.AssignmentId,
+    options: new CloseJobOptions()  // this is optional
+    {
+        CloseTime = DateTimeOffset.UtcNow.AddSeconds(2), // this will mark the job as closed after 2 seconds
+        Note = $"Job has been marked to close in the future by {worker.Value.Id} at {DateTimeOffset.UtcNow}"
+    });
+Console.WriteLine($"Job has been marked to close: {closeJob.GetRawResponse().Status == 202}"); // You'll received a 202 in that case
+
+await Task.Delay(TimeSpan.FromSeconds(2));
+*/
```
[the remark this answers is not in the document]
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md:261]: It shows an additional way to achieve the same action of closing a job.

File: sdk/communication/Azure.Communication.JobRouter/samples/ClassificationPolicyCrud.md [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+                }),
+            new ConditionalWorkerSelectorAttachment(
+                condition: new ExpressionRule("If(job.HighPriority = \"true\", true, false)"),
+                labelSelectors: new List<WorkerSelector>()
+                {
+                    new WorkerSelector("Skill_O365_Lvl", LabelOperator.GreaterThanEqual, new LabelValue(10))
+                })
+        }
+    });
+    */
```
[the remark this answers is not in the document]
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/ClassificationPolicyCrud.md:89]: The note attached to the comment is important in this case. This shows that while creating a `classification policy` a user can pick and choose which all properties of the policy that they want to define.

File: sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md [hunk 22]
```diff
@@ -0,0 +1,10 @@
+## 1.0.0-beta.1 (2021-10-20)
```
[the remark this answers is not in the document]
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md]: Removed release date. Changed to Unreleased

File: sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs (hunk 1 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs:79]: Done

File: sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs (hunk 2 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs:84]: Done

File: sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs (hunk 3 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs:100]: Done

File: sdk/communication/Azure.Communication.JobRouter/src/autorest.md (hunk 4 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/src/autorest.md:16]: The design implemeted here is that the subset of the models (everything used as input) should live under `Azure.Communication.JobRouter` and another subset lives under `Azure.Communication.Models` (everything used as output). The number of input models is greater than output models hence  `model-namespace: false`

File: sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md (hunk 5 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/CHANGELOG.md]: Done

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 8 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md]: This has been updated to be injected with snippet

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 10 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md]: This has been updated to not use `var` in all samples

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 11 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md]: This has been fixed

File: sdk/communication/Azure.Communication.JobRouter/samples/ExceptionPolicyCrud.md (hunk 12 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/ExceptionPolicyCrud.md]: This has been updated. Markdown formatting has been removed from code snippets

File: sdk/communication/Azure.Communication.JobRouter/samples/JobQueueCrud.md (hunk 13 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/JobQueueCrud.md]: This has been updated

File: sdk/communication/Azure.Communication.JobRouter/samples/README.md (hunk 14 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/README.md]: This has been removed

File: sdk/communication/Azure.Communication.JobRouter/samples/README.md (hunk 15 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/README.md]: Removed blank line

File: sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/ClassificationPolicyLiveTests.cs (hunk 16 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/ClassificationPolicyLiveTests.cs]: This has been removed

File: sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/DistributionPolicyLiveTests.cs (hunk 17 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/tests/RouterClients/DistributionPolicyLiveTests.cs]: This was a miss. Have been uncommented

File: sdk/communication/Azure.Communication.JobRouter/tests/Samples/ClassificationPolicyCrudOps.cs (hunk 18 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/tests/Samples/ClassificationPolicyCrudOps.cs:24]: This has been updated

File: sdk/communication/ci.yml (hunk 19 above)
Comment (sarkar-rajarshi) [reply sdk/communication/ci.yml:28]: This has been added

File: sdk/communication/Azure.Communication.JobRouter/README.md [hunk 23]
```diff
@@ -0,0 +1,300 @@
...
+```
+
+### Create a JobRouter Client
+
+This will allow you to interact with the JobRouter Service
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_CreateClient
+RouterClient routerClient = new RouterClient("<< CONNECTION STRING >>");
+RouterAdministrationClient routerAdministrationClient = new RouterAdministrationClient("<< CONNECTION STRING >>");
+```
+
```
Comment (jsquire) [inline sdk/communication/Azure.Communication.JobRouter/README.md:37]:
```suggestion

```

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 9 above)
Comment (jsquire) [reply sdk/communication/Azure.Communication.JobRouter/README.md:157]: Your tests should either take that dependency or reference the Event Grid snippet directly here.   In the latter case, I'd suggest adding a note in the Event Grid source to note the snippet is used outside that scope.

The overall goal is to ensure that all code that appears in .NET SDK samples is compiled/tested and cannot rot over time.

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 23 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md:37]: fixed

File: sdk/communication/Azure.Communication.JobRouter/src/Models/CreateExceptionPolicyOptions.cs [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+        /// The Id of this policy.
+        /// </summary>
+        public string ExceptionPolicyId { get; set; }
+
+        /// <summary>
+        /// A dictionary collection of exception rules on the exception policy. Key is the Id of each exception rule.
+        /// </summary>
+#pragma warning disable CA2227 // Collection properties should be read only
+        public IDictionary<string, ExceptionRule> ExceptionRules { get; set; } =
+#pragma warning restore CA2227 // Collection properties should be read only
```
[the remark this answers is not in the document]
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/src/Models/CreateExceptionPolicyOptions.cs:41]: Based on our conversation, this will be brought up during SDK review to obtain feedback from arch board regarding user experience and will take action accordingly.

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 9 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md]: This has been updated to be injected with a snippet

File: sdk/communication/Azure.Communication.JobRouter/samples/ClassificationPolicyCrud.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/ClassificationPolicyCrud.md]: The commented code block has been removed.

File: sdk/communication/Azure.Communication.JobRouter/README.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/README.md]: This has been uncommented

**Review by jsquire — APPROVED** [review]
General high-level patterns look good.   I didn't deep dive into the communication code itself.    Approving, but I'd like @ShivangiReja to be the authoritative vote.

**Review by ShivangiReja — APPROVED** [review]
LGTM
